scripts/variantstore/wdl/extract/create_variant_annotation_table.py

```diff
@@ -234,33 +235,37 @@ def make_annotated_json_row(row_position, row_ref, row_alt, variant_line, transc
-      # And then anything that did not match will get added at the end of the list (sorted alpha)
-      ordered_significance_values = []
-      for value in significance_ordering:
-        if value in significance_values:
-          ordered_significance_values.append(value) # this adds the id to the end of the list
-      values_not_accounted_for = list(set(significance_values).difference(ordered_significance_values))
-      values_not_accounted_for.sort()
-      ordered_significance_values.extend(values_not_accounted_for) # add any values that aren't in significance_ordering to the end
-      row["clinvar_id"] = clinvar_ids # array
-      row["clinvar_classification"] = ordered_significance_values # special sorted array
-      updated_dates.sort(key=lambda date: datetime.strptime(date, "%Y-%m-%d")) # note: method is in-place, and returns None
-      row["clinvar_last_updated"] = updated_dates[-1] # most recent date
-      row["clinvar_phenotype"] = sorted(phenotypes) # union of all phenotypes
+      # Note that inside the clinvar array, are multiple objects that may or may not be the one we are looking for. We check by making sure the ref and alt are the same
+      for clinvar_obj in clinvar_objs:
+         # get only the clinvar objs with right variant and the id that starts with RCV
+         if (clinvar_obj.get("refAllele") == var_ref) & (clinvar_obj.get("altAllele") == var_alt) & (clinvar_obj.get("id")[:3] == "RCV"):
+           clinvar_ids.append(clinvar_obj.get("id"))
+           significance_values.extend([x.lower() for x in clinvar_obj.get("significance")])
+           updated_dates.append(clinvar_obj.get("lastUpdatedDate"))
+           phenotypes.extend(clinvar_obj.get("phenotypes"))
+      if len(clinvar_ids) > 0:
+        ordered_significance_values = []
+        # We want to collect all the significance values and order them by the significance_ordering list
+        # So I will loop through the significance_ordering values and check for matching values in the significance_values list and put them in a new list
+        # And then anything that did not match will get added at the end of the list (sorted alpha)
+        for value in significance_ordering:
+          if value in significance_values:
+            ordered_significance_values.append(value) # this adds the id to the end of the list
+        values_not_accounted_for = list(set(significance_values).difference(ordered_significance_values))
+        values_not_accounted_for.sort() # alphabetize this so it is deterministic
+        ordered_significance_values.extend(values_not_accounted_for) # add any values that aren't in significance_ordering to the end
+        row["clinvar_id"] = clinvar_ids # array
+        row["clinvar_classification"] = ordered_significance_values # special sorted array
+        updated_dates.sort(key=lambda date: datetime.strptime(date, "%Y-%m-%d")) # note: method is in-place, and returns None
```

Comment:
thank you for the comment, I forget that every time I've been away from Python for a while and it never fails to bite me 🙂